huaweicloud/utils/utils.go

```diff
@@ -259,3 +259,41 @@ func EncodeBase64IfNot(str string) string {
 	}
 	return str
 }
+
+// This function compares whether there is a containment relationship between two maps, that is,
+// whether map A (rawMap) contains map B (filterMap).
+//   Map A is {'foo': 'bar'} and map B is {'foo': 'dor'} will return empty map and false.
+//   Map A is {'foo': 'bar,dor'} and map B is {'foo': 'dor'} will return Map A and true.
+// The value 'bar,for' means that the object value can be either 'bar' or 'dor'.
+// Note: There is no spaces before and after the delimiter (,).
+func HasMapContains(rawMap map[string]string, filterMap map[string]interface{}) (map[string]string, bool) {
+	if len(filterMap) < 1 {
+		return rawMap, true
+	}
+	if len(filterMap) > 0 && len(rawMap) < 1 {
+		return nil, false
+	}
+
+	isContain := true
+	for key, value := range filterMap {
+		isContain = isContain && hasMapContain(rawMap, key, value.(string))
+	}
+
+	if isContain {
+		return rawMap, true
+	}
+	return nil, false
+}
+
+func hasMapContain(rawMap map[string]string, filterKey, filterValue string) bool {
```

Comment:
Map A is {'foo': 'bar'} and map B is {'foo': ''} will return Map A and true.
The return we expect should be empty map and false.

Reply:
The description of the function `HasMapContians` is wrong, the value of rawMap cannot contain a comma (,), but it should be contained in filterMap

Reply:
An empty value means that only the specified key is matched.